src/uncurryN.js

```diff
@@ -0,0 +1,43 @@
+var _curry2 = require('./internal/_curry2');
+var _slice = require('./internal/_slice');
+var arity = require('./arity');
+var is = require('./is');
+
+
+/**
+ * Returns a function of arity `n` from a (manually) curried function.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig Number -> (a -> b) -> (a -> c)
+ * @param {Number} length The arity for the returned function.
+ * @param {Function} fn The function to uncurry.
+ * @return {Function} A new function.
+ * @see R.curry
+ * @example
+ *
+ *      var addFour = function(a) {
+ *        return function(b) {
+ *          return function(c) {
+ *            return function(d) {
+ *              return a + b + c + d;
+ *            };
+ *          };
+ *        };
+ *      };
+ *
+ *      var uncurriedAddFour = R.uncurryN(4, addFour);
+ *      curriedAddFour(1, 2, 3, 4); //=> 10
+ */
+module.exports = _curry2(function uncurryN(length, fn) {
+  return arity(length, function() {
+    var idx = 0;
+    var value = fn;
+    while (idx < length && is(Function, value)) {
```

Comment:
This probably could be done with metadata.  But it would also work the other way, and that might get a bit more crazy:

```javascript
var fn1 = uncurry(fn);
var fn2 = curry(fn1);
fn === fn2; //=> true
```

But I don't think such invariants are actually all that important.  The point of `uncurry`/`recurry`, or whatever we end up calling it, whether or not the result actually ends up curried, is that it could take a function that was manually curried to depth `n` and return an arity-n function with the same behaviour.  This implementation  has the nice advantage that it would also perform the same service for an already arity-n function or one that was curried through a `curry` function like Ramda's, meaning that it could be used fairly widely, but that's not central.  And actual function identity is still further from the core.